### PR TITLE
Feature/cakephp auth plugins

### PR DESCRIPTION
**Pull request by rochamarcelo**
Using cakephp/authentication and cakephp/authorization. Moved social logic to auth plugin.

### CODE REVIEW

File: src/Controller/Traits/LinkSocialTrait.php [hunk 1]
```diff
@@ -57,74 +49,35 @@ public function linkSocial($alias = null)
...
-
-        try {
-            $data = $this->_mapSocialUser($alias, $data);
-
-            $user = $this->getUsersTable()->get($this->Auth->user('id'));
+            $data = $server->getUser($this->request);
+            $mapper = new MapUser();
+            $data = $mapper($server, $data);
+            $identity = $this->request->getAttribute('identity');
+            $identity = isset($identity) ? $identity : [];
```
[the remark this answers is not in the document]
Comment (rochamarcelo) [reply src/Controller/Traits/LinkSocialTrait.php]: We can't use Hash::get to get an attribute.

[the remark this answers is not in the document]
Comment (rochamarcelo) [reply src/Controller/Traits/LinkSocialTrait.php]: And this does not seems better $identity = Hash::get(
                $this->request->getAttributes(),
                'identity',
                []
            );

[the remark this answers is not in the document]
Comment (steinkel) [reply src/Controller/Traits/LinkSocialTrait.php]: nop, I was going to suggest replacing all Hash::get usage later... so we're good here

File: Docs/Documentation/SocialAuthenticate.md [hunk 2]
```diff
@@ -95,3 +97,70 @@ $this->addBehavior('CakeDC.Users/Social', [
...
+
+Social Indentifier
+------------------
+The social identifier "CakeDC/Users.Social", works with data provider by both social authenticator,
+it is responsible to find or create a user registry for the request social user data.
+By default it fetch user data with finder 'all', but you can use a one if you need. Add this to your
+Application class, after CakeDC/Users Plugin is loaded.
+```
+    $identifiers = Configure::read('Auth.Identifiers');
+    $identifiers['CakeDC/Users.Social']['authFinder'] = 'customSocialAuth';
```
[the remark this answers is not in the document]
Comment (rochamarcelo) [reply Docs/Documentation/SocialAuthenticate.md]: Each identifier has its own configuration. I updated the default config to use 'active' finder. Password login allow username or email.

File: src/Controller/Component/LoginComponent.php [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+     * Handle login, if success redirect to 'AuthenticationComponent.loginRedirect' or show error
+     *
+     * @param bool $errorOnlyPost should handle failure only on post request
+     * @param bool $redirectFailure should redirect on failure?
+     * @return \Cake\Http\Response|null
+     */
+    public function handleLogin($errorOnlyPost, $redirectFailure)
+    {
+        $request = $this->getController()->getRequest();
+        $result = $request->getAttribute('authentication')->getResult();
```
[the remark this answers is not in the document]
Comment (rochamarcelo) [reply src/Controller/Component/LoginComponent.php]: Added check

File: src/Utility/UsersUrl.php [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+     * Get an user action url
+     *
+     * @param string $action user action
+     * @param array $extra extra url attributes
+     *
+     * @return array
+     */
+    public function actionUrl($action, $extra = [])
+    {
+        $prefix = false;
```
Comment (steinkel) [inline src/Utility/UsersUrl.php]: prefix false vs null?

**Pull request merged**